src/python/CMSSpark/data_aggregation.py

```diff
@@ -0,0 +1,385 @@
+from CMSSpark.spark_utils import spark_context, print_rows, split_dataset
+from CMSSpark.utils import elapsed_time
+from CMSSpark.data_collection import  yesterday, short_date_string, long_date_string, output_dataframe, run_query
+from pyspark.sql.functions import desc
+
+class OptionParser():
+    def __init__(self):
+        "User based option parser"
+        desc = "Spark script to process DBS + [AAA, CMSSW, EOS, JM] metadata"
+
+        self.parser = argparse.ArgumentParser(prog='PROG', description=desc)
+
+        self.parser.add_argument("--inst", action="store",
+            dest="inst", default="global", help='DBS instance on HDFS: global (default), phys01, phys02, phys03')
+        self.parser.add_argument("--date", action="store",
+            dest="date", default="", help='Select data for specific date (YYYYMMDD)')
+        self.parser.add_argument("--yarn", action="store_true",
+            dest="yarn", default=False, help="Run job on analytics cluster via yarn resource manager")
+        self.parser.add_argument("--verbose", action="store_true",
+            dest="verbose", default=False, help="Verbose output")
+        self.parser.add_argument("--fout", action="store",
+            dest="fout", default="", help='Output directory path')
+
+
+def run_agg_jm(date, fout, ctx, sql_context, verbose=False):
+
+    if verbose:
+        print 'Starting JobMonitoring part'
+
+    # Convert date
+    date = long_date_string(date)
+
+    # Create JobMonitoring tables in sql_context
+    jm_df = jm_tables(ctx, sql_context, date=date, verbose=verbose)
+
+    if verbose:
+        print 'Found ' + str(jm_df['jm_df'].count()) + ' records in JobMonitoring stream'
+
+    # - site name                +
+    # - dataset name             +
+    # - number of access (nacc)  +
+    # - distinct users           +
+    # - stream: crab             +
+
+    cols = ['SiteName AS site_name',
+            'dataset_name',
+            'count(dataset_name) AS nacc',
+            'count(distinct(UserId)) AS distinct_users',
+            '\"crab\" as stream']
+
+    # Build a query with "cols" columns
+    query = ("SELECT %s FROM jm_df "\
+             "JOIN f_b_s_df ON f_b_s_df.file_name = jm_df.FileName "\
+             "GROUP BY jm_df.SiteName, dataset_name") % ','.join(cols)
+
+    result = run_query(query, sql_context, verbose)
+
+    result = result.sort(desc("nacc"))
+
+    if verbose:
+        print 'Finished JobMonitoring part (output is ' + str(result.count()) + ' records)'
+
+    return result
+
+
+def run_agg_eos(date, fout, ctx, sql_context, verbose=False):
+
+    if verbose:
+        print 'Starting EOS part'
+
+    date = short_date_string(date)
+
+    # Create EOS tables in sql_context
+    eos_df = eos_tables(sql_context, date=date, verbose=verbose)
+
+    if verbose:
+        print 'Found ' + str(eos_df['eos_df'].count()) + ' records in EOS stream'
```

Comment:
use new print function syntax and use format option instead of concatenaning strings, e.g.
print("Found %s records in EOS string" % eos_df['eos_df'].count())